experimental/dds/tree2/src/feature-libraries/sequence-field/rebase.ts

```diff
@@ -133,95 +132,59 @@ function rebaseMarkList<TNodeChange>(
-					} else {
-						factory.pushOffset(getOutputLength(baseMark));
-					}
-				} else if (baseMark.transientDetach === undefined) {
-					factory.pushOffset(getOutputLength(baseMark));
-				}
-				lineageRecipients.length = 0;
-				lineageEntries.length = 0;
-			}
-		} else {
-			assert(
-				getInputLength(baseMark) === getInputLength(currMark),
-				0x4f6 /* The two marks should be the same size */,
-			);
-
-			const rebasedMark = rebaseMark(
-				currMark,
-				baseMark,
-				baseRevision,
-				baseIntention,
-				rebaseChild,
-				moveEffects,
-				nodeExistenceState,
-			);
+		assert(
+			getInputLength(baseMark) === getInputLength(currMark),
+			0x4f6 /* The two marks should be the same size */,
+		);
 
-			// Note that we first add lineage for `baseMark` to `lineageRecipients`, then handle adding lineage to `rebasedMark`,
-			// then add `baseMark` to `lineageEntries` so that `rebasedMark` does not get an entry for `baseMark`.
-			// `rebasedMark` should already have a detach event for `baseMark`.
-			if (markEmptiesCells(baseMark)) {
-				assert(isDetachMark(baseMark), 0x709 /* Only detach marks should empty cells */);
-				addLineageToRecipients(
-					lineageRecipients,
-					baseIntention,
-					baseMark.id,
-					baseMark.count,
-				);
-			}
+		const rebasedMark = rebaseMark(
+			currMark,
+			baseMark,
+			baseRevision,
+			baseIntention,
+			rebaseChild,
+			moveEffects,
+			nodeExistenceState,
+		);
 
-			if (areInputCellsEmpty(rebasedMark)) {
-				handleLineage(rebasedMark, lineageRecipients, lineageEntries, baseIntention);
-			}
-			factory.push(rebasedMark);
+		// Note that we first add lineage for `baseMark` to `lineageRecipients`, then handle adding lineage to `rebasedMark`,
+		// then add `baseMark` to `lineageEntries` so that `rebasedMark` does not get an entry for `baseMark`.
+		// `rebasedMark` should already have a detach event for `baseMark`.
+		if (markEmptiesCells(baseMark)) {
+			assert(isDetachMark(baseMark), 0x709 /* Only detach marks should empty cells */);
+			addLineageToRecipients(lineageRecipients, baseIntention, baseMark.id, baseMark.count);
+		}
 
-			if (markEmptiesCells(baseMark)) {
-				assert(isDetachMark(baseMark), 0x70a /* Only detach marks should empty cells */);
-				lineageEntries.push({ id: baseMark.id, count: baseMark.count });
-			} else {
-				lineageRecipients.length = 0;
-				lineageEntries.length = 0;
-			}
+		if (areInputCellsEmpty(rebasedMark)) {
+			handleLineage(rebasedMark, lineageRecipients, lineageEntries, baseIntention);
+		}
+		factory.push(rebasedMark);
+
+		if (markEmptiesCells(baseMark)) {
+			assert(isDetachMark(baseMark), 0x70a /* Only detach marks should empty cells */);
+			lineageEntries.push({ id: baseMark.id, count: baseMark.count });
+		} else if (!areOutputCellsEmpty(baseMark)) {
+			lineageRecipients.length = 0;
+			lineageEntries.length = 0;
 		}
 	}
 
 	return factory.list;
 }
 
+function generateNoOpWithCellId<T>(mark: Mark<T>, revision?: StableId): NoopMark {
```

Comment:
Maybe add a comment or rename this to make it more clear what this is for, namely, creating a noop mark which targets the same cells as the input mark.

Reply:
Added doc comments for this function